packages/lesswrong/server/callbacks/dialogueMatchPreferenceCallbacks.ts

```diff
@@ -190,8 +190,7 @@ const welcomeMessage = (formDataSourceUser: MatchPreferenceFormData, formDataTar
   return `<div>${messagesCombined}</div>`
 }
 
-getCollectionHooks("DialogueMatchPreferences").createBefore.add(async function GenerateDialogue ( userMatchPreferences, { context, currentUser } ) {
-  const { dialogueCheckId } = userMatchPreferences;
+const getReciprocalMatchPreference = async (context: ResolverContext, currentUser: DbUser | null, dialogueCheckId: string) => {
   const dialogueCheck = await context.loaders.DialogueChecks.load(dialogueCheckId);
 
   // This shouldn't ever happen
```

Comment:
FYI the line under this looks like it's using `??` instead of the `||` operator, where it needs to be using `||` (since it's checking whether any of the conditions evaluate to `true`, and `??` won't do anything when operating on booleans, which those will be since they're all either coerced with `!` or coming out of equality comparisons).

Reply:
This is code written by Oli.